prisma/schema.prisma

```diff
@@ -16,12 +16,27 @@ model User {
   userName String         @unique
   password String
   score    Int
+  role     Role           @default(USER)
   games    UsersOnGames[]
 }
 
+model Admin {
+  id       Int    @id @default(autoincrement())
+  userName String @unique
+  password String
+  role     Role   @default(ADMIN)
+}
```

Comment:
I have simply created the admin to manage the games, for example, he would be the only one that could delete games. I'm too sure if I'm gonna implement that, but the admin would be the only one who can access a list of all the games. but it was more for exercising roles.. anyway if you want me to delete it I will =)

Reply:
Ah, I hear you. So the admin itself is just a record in the User table, when you create a user you'll end up with an object like this:
```js
{
    id: 1
    userName: "LanaIsGreat",
    password: "asdfgasd0f9834f0234hf09h0",
    score: 420,
    role: "USER"
}
```

You can use the same model for admins, so if you created that above user and later decided "Y'know what? I want to make that user an admin", you can just go to the database and change that users role from "USER" to "ADMIN", then that record in the database would look like:

```js
{
    id: 1
    userName: "LanaIsGreat",
    password: "asdfgasd0f9834f0234hf09h0",
    score: 420,
    role: "ADMIN"
}
```

It's the same data, the same record in the same table. If you had a separate admin model and you decided you wanted to make LanaIsGreat an admin, you'd have to create a brand new Admin record for that person and then delete the old record in the User table.

The reason you want to keep to the same model and use a Role property is so that you can use the same routes and controllers and just throw in a simple
```js
if (user.role === "ADMIN") {
   // This user is an admin, let them do some fun stuff
}
```

By having a separate Admin model, you would have to create a completely new set of routes and controllers for it - hugely increasing your code size without giving much benefit